mmf/datasets/databases/readers/feature_readers.py

```diff
@@ -103,16 +104,18 @@ def read(self, image_feat_path):
 class CHWFeatureReader:
     def __init__(self, max_features=None):
         self.max_features = max_features
+        if self.max_features:
```

Comment:
Seems very hacky specifically for grid features, would it make sense to create a new feature reader (though TBH it doesn't matter as this is not used anywhere else lol)